### ISSUE TITLE
Support for package upgrade 

### ISSUE THREAD

**Issue by mauhiz** [transcode-open/apt-cyg#22]
Is there a way to upgrade packages?

e.g. `apt-cyg upgrade bash` or `apt-cyg upgrade --all`

**Comment by gpakosz** [comment]
That would indeed be very useful

**Comment by CMCDragonkai** [comment]
Yea I was missing this too?

**Comment by SandHilt** [comment]
+1

**Comment by E3V3A** [comment]
Running `setup-x86.exe` or `setup-x86_64.exe`, automatically updates all installed packages except those that has been manually compiled and installed. But, yeah, this need to be said somewhere in the `apt-cyg --help` text and I'm not sure *apt-cyg* is or can be run that way.

**Comment by svnpenn** [comment]
I removed @E3V3A comment because I dont ever want to hear about `setup-*.exe` on this repo. That crap program is the exact reason apt-cyg was created. I do plan on implementing package upgrade, but if someone wants to submit a patch I will consider it.

**Comment by gpakosz** [comment]
@svnpenn you **do** or **do not** plan to support package upgrade?

**Comment by svnpenn** [comment]
@gpakosz of course, I want to have package upgrade. Of course I can write it as well. However as I do this project for free in my spare time, it is low priority.

**Comment by gpakosz** [comment]
@svnpenn sure, thing. I'm not pushing you. I just wasn't sure whether your last sentence contained a typo or not.

**Comment by banholzer** [comment]
Hey, I'd love this feature, too. 
I'd like to help, but I don't know Go. Need to look at Go before.

**Comment by svnpenn** [comment]
@banholzer I will consider patches for this feature. Also, your wiki entries are not welcome, please remove them.

**Comment by E3V3A** [comment]
@svnpenn 

>I removed @E3V3A comment because I dont ever want to hear about setup-*.exe on this repo. That crap program is the exact reason apt-cyg was created. I do plan on implementing package upgrade, but if someone wants to submit a patch I will consider it.

That was totally unnecessary since I was doing you favor, by telling people how they can do that themselves and thus avoiding to have to post about it in this topic as a bug. But that wasn't welcome I guess. In addition I find it a lame excuse to say you don't have time, since this issue was posted already one year ago. (And yes I support your tool and yes I think the Cygwin package management sucks too.)

**Comment by svnpenn** [comment]
@E3V3A no, you do not support my tool. Last time I check I have not received a donation from you or anyone else regarding this project, so you can fuck right off. This feature will be implemented **when I get damn good and ready**, or when someone submits a patch and I approve it. If you do not like it, you can go to hell. This is open source so you better get used to it.

**Comment by svnpenn** [comment]
If anyone wants to really support this issue, please see this link, thank you.

http://bountysource.com/issues/4807368-support-for-package-upgrade

**Comment by svnpenn** [comment]
If anyone wants to really support this issue, please see this link, thank you.

[![Bountysource](https://www.bountysource.com/badge/issue?issue_id=4807368)](https://www.bountysource.com/issues/4807368-support-for-package-upgrade?utm_source=4807368&utm_medium=shield&utm_campaign=ISSUE_BADGE)

**Comment by nitrocode** [comment]
Exactly what I wanted. Yes, adding this functionality would be useful.

Correct me if I'm wrong, but I believe you could use this statement:

    apt-cyg update && apt-cyg list | while read f; do apt-cyg install $f -y; done

**Comment by svnpenn** [comment]
@nitrocode no. If you do that, it just say everything is already installed.

**Comment by nitrocode** [comment]
@svnpenn, how about this?

    export cygexepath="/cygdrive/c/cygwin64/setup-x86_64.exe"
    apt-cyg update && apt-cyg list | while read f; do $cygexepath -q -P $f; done

**Comment by fnk0c** [comment]
Bash is not my "best language", but I could also help, if I knew how would this function work.

**Comment by brcolow** [comment]
This would be a really killer feature imho. Thanks for your work thus far.

**Comment by brcolow** [comment]
@svnpenn do you have any pointers on how such a feature could be implemented?

**Comment by svnpenn** [comment]
@brcolow I am sorry to say that I was removed as maintainer for this repo because of some drama. However if you like you could resubmit this issue at my new repo for further discussion:

[github.com/svnpenn/sage](//github.com/svnpenn/sage)

**Comment by svnpenn** [comment]
https://github.com/svnpenn/sage/issues/1

**Comment by keithdtyler** [comment]
@svnpenn never explains how to use his sage fixes from apt-cyg. I don't have sage, and I'm not looking at the sage issues page, I have apt-cyg and I'm on the apt-cyg page.

**Comment by damon-kwok** [comment]
+1

**Comment by techlunacy** [comment]
hi all i have moved onto the @kou1okada fork of this that has the upgrade option
[https://github.com/kou1okada/apt-cyg](https://github.com/kou1okada/apt-cyg)
there may be more uptodate forks out there, so best have a look

**Comment by cup** [comment]
https://github.com/cup/sage/issues/20
